src/rules/noRequireImportsRule.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "require() style import is forbidden for module: ";
```

Comment:
I don't think you need to provide the name of the module they're trying to import in the error message. Imports are almost always grouped at the top of a file and the user will get position information about the failure anyway, so it should be easy to locate. Adding this feature is just more work for us.